src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -495,6 +495,17 @@ private unsafe bool StartWithCreateProcess(ProcessStartInfo startInfo)
                         startupInfo.dwFlags = Interop.Advapi32.StartupInfoOptions.STARTF_USESTDHANDLES;
                     }
 
+                    if (startInfo.WindowStyle != ProcessWindowStyle.Normal)
+                    {
+                        startupInfo.wShowWindow = startInfo.WindowStyle switch
+                        {
+                            ProcessWindowStyle.Minimized => (short)Interop.Shell32.SW_SHOWMINIMIZED,
+                            ProcessWindowStyle.Maximized => (short)Interop.Shell32.SW_SHOWMAXIMIZED,
+                            _ => (short)Interop.Shell32.SW_HIDE,
+                        };
+                        startupInfo.dwFlags |= Interop.Advapi32.StartupInfoOptions.STARTF_USESHOWWINDOW;
+                    }
+
                     // set up the creation flags parameter
                     int creationFlags = 0;
                     if (startInfo.CreateNoWindow) creationFlags |= Interop.Advapi32.StartupInfoOptions.CREATE_NO_WINDOW;
```

Comment:
@iSazonov was suggesting that this combined with a WindowStyle != Normal should throw. But that would be a breaking change.

I lean more towards letting the OS decide whether is an error or not. i.e: to do nothing. I just wanted to point out that this is a deliberate decision.

Reply:
I just saw your comment https://github.com/dotnet/runtime/issues/81681#issuecomment-1451353923 talking about this.